DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionTaskDelegate+Tracking.swift

```diff
@@ -0,0 +1,28 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+private var firstPartyHostsKey: UInt8 = 41
+private var coreKey: UInt8 = 43
```

Comment:
And also on keys. I'm not entirely sure if we're doing this correctly.

From what I quickly read we need to provide keys that are "guaranteed to be constant, unique, and scoped for use within getters and setters"
https://stackoverflow.com/questions/29487330/proper-key-format-for-objc-setassociatedobject

boils down to either using static or #selector

We probably want to change this in all the other places in the project.
cc @ncreated

Reply:
although now when I look at it again - it's probably fine since it's using private global var 🤔

Reply:
we are not using this outside delegate, I don't see any issue unless someone do the same thing for delegate which is very unlikely and we use a constant other than easy to start.